lib/OpenQA/WebAPI/Controller/API/V1/Comment.pm

```diff
@@ -0,0 +1,95 @@
+# Copyright (C) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::WebAPI::Controller::API::V1::Comment;
+use Date::Format;
+use Mojo::Base 'Mojolicious::Controller';
+use OpenQA::Utils;
+use OpenQA::IPC;
+
+sub comments {
+    my ($self) = @_;
+
+    my $table = $self->param('job_id') ? "Jobs" : "JobGroups";
+    my $id = int($self->param('job_id') // $self->param('group_id'));
+    my $job = $self->app->schema->resultset($table)->find($id);
+    if (!$job) {
+        $self->render(json => {error => "Job $id does not exist"}, status => 404);
```

Comment:
Exactly - as I mentioned my version is easier to read. I think if you insist on removing code duplication, the better approach is a function.
```
sub obj_comments {
  my ($self, $param, $table, $label) = @_;
  my $id = int($self->param($param));
  my $obj = $self->app->schema->resultset($table)->find($id);
  if (!$obj) {
    $self->render(json => {error => "$label $id does not exist"}, status => 404);
    return;
  }
  return $obj->comments;
}

sub comments {
  if ($self->param('job_id')) {
    return $self->obj_comments('job_id', 'Jobs', 'Job');
  } else {
    return $self->obj_comments('group_id', 'JobGroups', 'Job group');
  }
}
```